Lambda-School-Labs/mydish-fe
# Feature/version history view

This PR styles the `VersionHistoryList` according to the design spec, which includes rendering the list within the `IndividualRecipe` view, rather than navigating to a new page.

**To test:** 
1. Press the "Other versions" link in a saved recipe.
2. Press on a version in the list. Does everything display correctly? Great!
3. Press "Other versions" again. Try looking through each version.
4. _Also_ try opening a recipe which you know doesn't have any other versions. Does the recipe have an "Other versions" link? If so, I didn't fix my bug* as well as I thought I did. If not, we're good.

_Note:_ When you open a recipe with no other versions, Expo may display the error below. This is because we're trying to fetch the versions for the recipe, but the recipe _has_ no versions, so the resource isn't found. The failed fetch lets us render the "Other versions" link conditionally. The error isn't a problem.

![Screen Shot 2020-03-25 at 4 34 56 PM](https://user-images.githubusercontent.com/50860480/77595555-b64fae80-6eb6-11ea-90a9-3a7ccbc03d5f.jpg)


*So, here's what the bug was. If you opened a recipe that had other versions, then went and opened a different recipe that _didn't_ have other versions, that recipe would have an "Other versions" link anyway, and when pressed it would open the version list of the _previous_ recipe. The  `resetAllVersionHistory` action I added in the final commit seems to take care of this properly. Do try to break it, though. See what happens.
